### PR TITLE
Support projects with Solidity and Vyper files

### PR DESCRIPTION
**Pull request by transferAndCall**
Fixes mixed-language environments to support compiling Solidity and Vyper in the same project.

### CODE REVIEW

**Comment by alcuadrado** [comment]
Ohh, you did it! thanks 😍

File: packages/hardhat-vyper/src/index.ts [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
 
 extendConfig((config) => {
   const defaultConfig = { version: "latest" };
   config.vyper = { ...defaultConfig, ...config.vyper };
 });
 
-task(TASK_COMPILE, async (_, { config, artifacts }) => {
+subtask(
+  TASK_COMPILE_GET_COMPILATION_TASKS,
+  async (): Promise<string[]> => {
```
Comment (alcuadrado) [inline packages/hardhat-vyper/src/index.ts:17]: This is a little picky, but when you override a task you can call the original one, so here I'd do

```ts
  async (_, _, runSuper): Promise<string[]> => {
    const otherTasks = await runSuper();
    return [...otherTasks, TASK_COMPILE_VYPER];
  }
```

This way the plugin is more robust and can interact with others that also override this task.

**Review by alcuadrado — COMMENTED** [review]
This looks good. Thanks for doing it, and adding a test.

I just left a minor comment.

File: packages/hardhat-vyper/src/index.ts (hunk 1 above)
Comment (transferAndCall) [reply packages/hardhat-vyper/src/index.ts:17]: Exactly what I was looking for.

**Comment by alcuadrado** [comment]
We'll release this early next week. We try not to do releases during the weekend.

**Pull request merged**